functions/initLease/src/index.ts

```diff
@@ -0,0 +1,193 @@
+import { APIGatewayEvent, Handler } from 'aws-lambda';
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocumentClient, QueryCommand, PutCommand, UpdateCommand } from '@aws-sdk/lib-dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { StatusCodes, ApiResponse, getEnv, Stage, MaxfriseErrorCodes } from '../../common';
+import { PROPERTY_STATUS, InitLeaseRequest, LandLord, Tenant, Responsebody } from "./types"
+
+const client = new DynamoDBClient({ region: "us-west-2" });
+const ddb = DynamoDBDocumentClient.from(client);
+
+export const handler: Handler<APIGatewayEvent, ApiResponse<InitLeaseRequest>> = async (event) => {
+  if (!event.body) return new ApiResponse<Responsebody>(StatusCodes.badRequest, { message: MaxfriseErrorCodes.missingInfo.message })
+  const body = JSON.parse(event.body) as InitLeaseRequest
+  const environment = getEnv(event.requestContext.stage)
+  const housesTableName = environment === Stage.PROD ? 'houses-prod' : 'houses'
+  const paymentJobsTableName = environment === Stage.PROD ? 'paymentJobs-prod' : 'paymentJobs'
+  let isUserOwner = false
+  let houseAvailable = false
+  let jobsCreated
+
+  try {
+    const result = await validateHouse(body.user, body.houseid, housesTableName)
+    isUserOwner = result.isUserOwner
+    houseAvailable = result.houseAvailable
+  } catch (error) {
+    console.error(error)
+    return new ApiResponse<Responsebody>(
+      MaxfriseErrorCodes.errorFromDynamo.code,
+      { message: MaxfriseErrorCodes.errorFromDynamo.message }
+    )
+  }
+
+  if (!isUserOwner) {
+    return new ApiResponse<Responsebody>(StatusCodes.badRequest, { message: 'The user is not the owner of the house' })
+  }
+
+  if (!houseAvailable) {
+    return new ApiResponse<Responsebody>(StatusCodes.badRequest, { message: 'The house is not avaible to be leased' })
+  }
+
+  try {
+    jobsCreated = await createJobs(
+      body.startDate,
+      body.term,
+      body.houseid,
+      body.rentAmount,
+      body.landlords,
+      body.tenants,
+      paymentJobsTableName
+    )
+  } catch (error) {
+    console.error(error)
+    return new ApiResponse<Responsebody>(
+      MaxfriseErrorCodes.errorFromDynamo.code,
+      { message: MaxfriseErrorCodes.errorFromDynamo.message }
+    )
+  }
+
+  try {
+    await updateHouseStatus(body.user, body.houseid, housesTableName)
+  } catch (error) {
+    console.error(error)
+    return new ApiResponse<Responsebody>(
+      MaxfriseErrorCodes.errorFromDynamo.code,
+      { message: MaxfriseErrorCodes.errorFromDynamo.message }
+    )
+  }
+
+  return new ApiResponse<Responsebody>(StatusCodes.ok,
+    {
+      message: "Lease initialized",
+      isUserOwner,
+      houseAvailable,
+      jobsCreated: jobsCreated.length
+    }
+  )
+};
+
+async function validateHouse(user: string, houseid: string, tableName: string) {
```

Comment:
I have no strong preference, I personally like to see all related logic in a single file as long as it is small and concise to not be jumping across files to understand the flow, but I could see how it may look a bit cleaner as well, will keep it like this for now, if we see a need latter it can be changed